presto-main/src/main/java/io/prestosql/server/protocol/QueryResultRows.java

```diff
@@ -82,15 +82,15 @@ private QueryResultRows(Session session, List<Type> types, List<Column> columns,
         this.currentPage = this.pages.pollFirst();
         this.supportsParametricDateTime = session.getClientCapabilities().contains(ClientCapabilities.PARAMETRIC_DATETIME.toString());
 
-        verify(this.types.size() == this.columns.size(), "columns and types sizes mismatch");
```

Comment:
should `types` be treated the same? ie also become Optional?

Reply:
Done.